### ISSUE TITLE
FormView is returning InvalidArrayOffset errors

### ISSUE THREAD

**Issue by VincentLanglet** [psalm/psalm-plugin-symfony#159]
When I add some field in `FormView->vars` I get no issues.
```
    public function buildView(FormView $view, FormInterface $form, array $options): void
    {
        $view->vars['all_fields'] = [];
        $view->vars['map'] = [];

        parent::buildView($view, $form, $options);
    }
```

But then the following code is returning an error `InvalidArrayOffset`
```
$myCustomType->buildView($view, $this->createStub(FormInterface::class), []);
$this->assertSame($expectedMap, $view->vars['map']);
```

It's because of https://github.com/psalm/psalm-plugin-symfony/blob/master/src/Stubs/common/forms/FormView.stubphp#L16 cc @zmitic

What would be the best solution to solve this @seferov  ?

**Comment by zmitic** [comment]
Reading ``$view->vars['map']`` is triggering an error, correct?

That's strange, I did put ``@psalm-suppress InvalidArrayOffset`` because of this. 

Can you paste full error message but give me few days to fix it, too busy right now.

**Comment by VincentLanglet** [comment]
I have a simpler example
```
    public function key(): string
    {
        $current = $this->current();

        return $current->vars['id'];
    }
```
`@psalm-suppress InvalidArrayOffset` psalm suppress does nothing.

I created https://github.com/psalm/psalm-plugin-symfony/pull/160

**Comment by zmitic** [comment]
@seferov I would like to discuss this issue. When I added another test after https://github.com/psalm/psalm-plugin-symfony/blob/master/tests/acceptance/acceptance/forms/FormView.feature#L46 like this:

```
$view->vars['attr']['placeholder'] = 'test'; // this is legit usage
```
I get an error:

```
| MixedArrayAssignment | Cannot access array value on mixed variable $view->vars['attr']['placeholder']                                                                                                                                   |
| PropertyTypeCoercion | $view->vars expects 'array{2: string, attr: array<array-key, mixed>, value: User|null}<string, mixed>',  parent type 'array{2: string, attr: mixed, random: stdClass, value: User|null}<string, mixed>' provided |
```

When I put back the original stub
```
@psalm-var array{value: ?T, attr: array<array-key, mixed>}
```

tests passed.

@VincentLanglet 

About this code:

```php
$myCustomType->buildView($view, $this->createStub(FormInterface::class), []);
$this->assertSame($expectedMap, $view->vars['map']);
```

Can you paste minimal reproducer?

**Comment by VincentLanglet** [comment]
@zmitic I made an error with 
```
@psalm-var array<string, mixed>&array{value: ?T, attr: array<array-key, mixed>, string}
```
It should be 
```
@psalm-var array<string, mixed>&array{value: ?T, attr: array<array-key, mixed>}
```

But I don't know if the order is important and 
```
@psalm-var array{value: ?T, attr: array<array-key, mixed>}&array<string, mixed>
```
Should be preferred.

**Comment by VincentLanglet** [comment]
@zmitic I think the form tests are not running because they are declared as `symfony-form` and not `symfony-common`

**Comment by VincentLanglet** [comment]
@zmitic In https://github.com/psalm/psalm-plugin-symfony/pull/166 I fixed the tests and the var phpdoc.
Psalm-trace is correctly returning mixed[] for `attr`
